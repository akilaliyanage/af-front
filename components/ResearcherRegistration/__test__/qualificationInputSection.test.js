import React  from 'react';
import ReactDOM from 'react-dom';
import {render, screen, cleanup} from '@testing-library/react'
import QualificationSection from '../qualificationInputSection'
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import expectExport from 'expect';

//clean up
afterEach(()=>cleanup())


//Test for successful rendering of the Qualification componant(Load without crash)
it('Chech Qualification succssful rendering',() =>{
    const div = document.createElement('div')
    let yearList = [2011 , 2012 , 2013 , 2014 , 2015]
    let UniList = ['Uni_1' , 'Uni_2' , 'Uni_3' , 'Uni_4']
    ReactDOM.render(
            <BrowserRouter>
                <QualificationSection UniList={UniList} yearList={yearList}/>
            </BrowserRouter>
    ,div)

})

//Test for successful rendering of the qualification componant - contain the select element
it('Qualification Componant renders correctly', () =>{
    let yearList = [2011 , 2012 , 2013 , 2014 , 2015]
    let UniList = ['Uni_1' , 'Uni_2' , 'Uni_3' , 'Uni_4']
    render(
            <BrowserRouter>
                <QualificationSection UniList={UniList} yearList={yearList}/>
            </BrowserRouter>
    )
    const qualificationContainer = screen.getByTestId('qualification-container')
    expectExport(qualificationContainer).toContainHTML("</datalist>")
})


//Compare snapshot of qualification componant for testing - any changes to the
//qualification componant will be detecteted after creating the snapshot
it('snapshot check for Publication', () =>{
    let yearList = [2011 , 2012 , 2013 , 2014 , 2015]
    let UniList = ['Uni_1' , 'Uni_2' , 'Uni_3' , 'Uni_4']
    const componant = renderer.create(
            <BrowserRouter>
                <QualificationSection UniList={UniList} yearList={yearList}/>
            </BrowserRouter>
    ).toJSON();

    expect(componant).toMatchSnapshot()
    
})

