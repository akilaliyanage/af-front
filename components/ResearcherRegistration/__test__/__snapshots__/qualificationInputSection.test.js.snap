// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot check for Publication 1`] = `
<div
  className="qualification-container"
>
  <div
    className="form-row-container qualification-row"
  >
    <div
      className="form-Element-container"
    >
      <input
        className="Researcher-form-input-long"
        id="degreeName"
        name="degreeName"
        type="text"
      />
      <label>
        Name of the Degree
      </label>
    </div>
  </div>
  <div
    className="form-row-container qualification-row"
    data-testid="qualification-container"
  >
    <div
      className="form-Element-container"
    >
      <input
        className="Researcher-form-input-long"
        id="UniName"
        list="uniList"
        name="UniName"
        type="text"
      />
      <datalist
        id="uniList"
      />
      <label>
        Name of the University
      </label>
    </div>
    <div
      className="form-Element-container"
    >
      <select
        className="Researcher-form-input-address-extras"
        id="qualiYear"
        name="qualiYear"
      >
        <option
          value={2011}
        >
          2011
        </option>
        <option
          value={2012}
        >
          2012
        </option>
        <option
          value={2013}
        >
          2013
        </option>
        <option
          value={2014}
        >
          2014
        </option>
        <option
          value={2015}
        >
          2015
        </option>
      </select>
      <label>
        Year
      </label>
    </div>
  </div>
</div>
`;
